conda-incubator/conda-store-ui
# Add ability to create environment from lockfile

This is a companion PR for https://github.com/conda-incubator/conda-store/pull/772.




## Description


This pull request: adds a new dropdown in the YAML window, which allows creating an env from a lockfile.

The supported lockfile type is https://conda.github.io/conda-lock/output/#unified-lockfile, which is also generated by conda-store as the lockfile build artifact.

## Pull request checklist



## Additional information


Where to get a lockfile for testing:
- create a standard conda-store environment
- copy the generated lockfile from the build artifacts
- create an env with this lockfile by selecting the lockfile specification type and pasting the copied lockfile.

Manual tests:

- [x] can create an environment using YAML -> lockfile specification type, deps and channels are updated on the GUI page on save
- [x] can edit lockfile environment, same
- [x] can edit a standard env and use lockfile type for the new revision
- [x] can edit a lockfile env and use specification type for the new revision
- [x] expected to not work: GUI is not updated without saving the env
- [x] expected to not work: lockfile is created from a GUI alone (not enough information in the GUI to create a lockfile entry)
- [x] expected to not work: no data is migrated when switching between specification types without saving

Future work:
- After this is merged, docs should be added to the conda-store repo describing the lockfile specification type. The supported lockfile type is https://conda.github.io/conda-lock/output/#unified-lockfile
- I had no time to create auto tests, some of the manual tests could be converted to auto tests.
